src/controllers/plugins/api/HTTP.cpp

```diff
@@ -0,0 +1,433 @@
+}
+
+int HTTPRequest::set_timeout_wrap(lua_State *L)
+{
+    auto ptr = HTTPRequest::getOrError(L, 1);
+    return ptr->set_timeout(L);
+}
+
+int HTTPRequest::set_timeout(lua_State *L)
+{
+    auto top = lua_gettop(L);
+    if (top != 1)
+    {
+        luaL_error(L, "HTTPRequest:set_timeout needs 1 argument (a number of "
+                      "milliseconds after which the request will time out)");
+        return 0;
+    }
+
+    int temporary = -1;
+    if (!lua::pop(L, &temporary))
+    {
+        luaL_error(L,
+                   "HTTPRequest:set_timeout failed to get timeout, expected a "
+                   "positive integer");
+        return 0;
+    }
+    if (temporary <= 0)
+    {
+        luaL_error(L,
+                   "HTTPRequest:set_timeout failed to get timeout, expected a "
+                   "positive integer");
+        return 0;
+    }
+    this->timeout_ = temporary;
+    return 0;
+}
+
+int HTTPRequest::finally_wrap(lua_State *L)
+{
+    auto ptr = HTTPRequest::getOrError(L, 1);
+    return ptr->finally(L);
+}
+
+int HTTPRequest::finally(lua_State *L)
+{
+    auto top = lua_gettop(L);
+    if (top != 1)
+    {
+        luaL_error(L, "HTTPRequest:finally needs 1 argument (a callback "
+                      "that takes nothing and doesn't return anything)");
+        return 0;
+    }
+    if (!lua_isfunction(L, top))
+    {
+        luaL_error(L, "HTTPRequest:finally needs 1 argument (a callback "
+                      "that takes nothing and doesn't return anything)");
+        return 0;
+    }
+    auto shared = this->pushPrivate(L);
+    lua_pushvalue(L, -2);
+    lua_setfield(L, shared, "finally");
+    return 0;
+}
+
+int HTTPRequest::set_payload_wrap(lua_State *L)
+{
+    auto ptr = HTTPRequest::getOrError(L, 1);
+    return ptr->set_payload(L);
+}
+
+int HTTPRequest::set_payload(lua_State *L)
+{
+    auto top = lua_gettop(L);
+    if (top != 1)
+    {
+        luaL_error(
+            L, "HTTPRequest:set_payload needs 1 argument (a string payload)");
+        return 0;
+    }
+
+    std::string temporary;
+    if (!lua::pop(L, &temporary))
+    {
+        luaL_error(L,
+                   "HTTPRequest:set_payload failed to get payload, expected a "
+                   "string");
+        return 0;
+    }
+    this->req_ =
+        std::move(this->req_).payload(QByteArray::fromStdString(temporary));
+    return 0;
+}
+
+int HTTPRequest::set_header_wrap(lua_State *L)
+{
+    auto ptr = HTTPRequest::getOrError(L, 1);
+    return ptr->set_header(L);
+}
+
+int HTTPRequest::set_header(lua_State *L)
+{
+    auto top = lua_gettop(L);
+    if (top != 2)
+    {
+        luaL_error(L, "HTTPRequest:set_header needs 2 arguments (a header name "
+                      "and a value)");
+        return 0;
+    }
+
+    QString value;
+    if (!lua::pop(L, &value))
+    {
+        luaL_error(L,
+                   "cannot get value (2nd argument of HTTPRequest:set_header)");
+        return 0;
+    }
+    QString name;
```

Comment:
Both `value` and `name` could be a `QByteArray` and a `NetworkRequest::header(const QByteArray&, const QByteArray&)` overload could be added (since that's the signature of `setRawHeader`).

Reply:
I added a new overload and changed this code to use it